buildSrc/src/main/java/org/mozilla/fenix/gradle/tasks/ApkSizeTask.kt

```diff
@@ -0,0 +1,126 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.gradle.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Paths
+
+/**
+ * Gradle task for determining the size of APKs and logging them in a perfherder compatible format.
+ */
+open class ApkSizeTask : DefaultTask() {
+    /**
+     * Name of the build variant getting built.
+     */
+    @Input
+    var variantName: String? = null
+
+    /**
+     * List of APKs that get build for the build variant.
+     */
+    @Input
+    var apks: List<String>? = null
+
+    @TaskAction
+    fun logApkSize() {
+        val apkSizes = determineApkSizes()
+        val json = buildPerfherderJson(apkSizes)
+
+        println("PERFHERDER_DATA: $json")
```

Comment:
^^ Did you mean to keep that in for debugging?

Reply:
If I understand it correctly then this is how perfherder will pick it up. It looks at the taskcluster logs and grabs those values from it.

Reply:
Ah OK! TIL.